src/main/kotlin/Executor.kt

```diff
@@ -49,24 +62,14 @@ class Executor(val engineConfig: EngineConfiguration) {
             } else if (test is SingleTest) {
 
                 val queryId = queryId.getAndAdd(1)
-                var stubId = 0
 
                 val success: ResultCase?
 
-                while (!usedStubs[stubId].tryLock()) {
-                    stubId += 1
-
-                    //Sleep for a bit when we've tried all stubs
-                    if (stubId >= numProcesses) {
-                        Thread.sleep(100)
-                        stubId = 0
-                    }
-                }
+                lock()
```

Comment:
I don't know if we actually need to lock anymore now that each query is a single GRPC request with the proto changes. I think the only reason we needed to lock before was that we first had to send components and then the query and we had to guarantee the order. I'm not sure about J-ecdar though but I would suspect that it will just blocks on the server side, so we still do not need to lock on our side.

This is probably a larger change, since we actually only want to spin up a single instance of Reveaal now and send it async queries instead of blocking, so I think we should leave it to a future PR